WFCORE-1314 Propertly handle java 9 version string

ctomc opened the pull request:
Fix the code to properly address new JDK 9 versioning as defined in http://openjdk.java.net/jeps/223

ctomc commented:
@jamezp can you take a look at this, as it does touch your launcher code.

dmlloyd commented on host-controller/src/test/java/org/jboss/as/host/controller/model/jvm/JvmOptionsBuilderUnitTestCase.java:
```diff
@@ -44,13 +44,17 @@
+                    vmVersion = Integer.valueOf(matcher.group(1));
+                } else {
+                    throw new RuntimeException("Unknown version of jvm " + vmVersionStr);
+                }
```
You have a regex here already, why do this with half regex and half string banging?

Just use a regex of "^(?:1\\.)?(\\d+)$" instead.

dmlloyd commented on launcher/src/main/java/org/wildfly/core/launcher/JavaVersionComparator.java:
```diff
@@ -59,40 +57,20 @@ public static int compareVersion(final String version1, final String version2) {
-                result.add(0);
-            }
+    private int getMajorVersion(String versionString) {
+        final String[] vs = versionString.split("\\.");
```
Now we're doing two splits here, to acquire literally 2 to 3 characters.  A regex is probably best here too.

ctomc replied:
I just kept it as it was in previous version of code. 
Makes sense to simplify it as much as possible.

ctomc commented:
@dmlloyd updated.

dmlloyd commented:
Thanks.

jamezp commented on host-controller/src/test/java/org/jboss/as/host/controller/model/jvm/JvmOptionsBuilderUnitTestCase.java:
```diff
@@ -43,14 +43,14 @@
-            vmVersion = 7;
+            vmVersion = 8;
         }
         JVM_MAJOR_VERSION = vmVersion;
```
This should just all be removed since we require Java 8 anyway.

ctomc replied:
makes sense

jamezp commented:
The launcher change isn't really needed. The original version works as is without the changes. That said if we're going to change I'd rather just have it removed. We require Java 8 now and if a user wants to use a new version of the launcher to run on older container they can add the JVM arguments manually.

jamezp commented:
FWIW this also relates to https://issues.jboss.org/browse/WFCORE-1189

ctomc commented:
@jamezp I've changed the JvmOptionsBuilderUnitTestCase to not have version check anymore.
Also removed JavaVersionComparator from launcher as you suggested.

ctomc commented:
@jamezp this got acked for merging to downstream, so we can remove 3.x tag
